extensions/2.0/Khronos/KHR_xmp/README.md

```diff
@@ -0,0 +1,337 @@
+| [Rational](https://github.com/adobe/xmp-docs/blob/master/XMPNamespaces/XMPDataTypes/CoreProperties.md#rational)            | String formatted as described in the XMP documentation |
+| [FrameRate](https://github.com/adobe/xmp-docs/blob/master/XMPNamespaces/XMPDataTypes/CoreProperties.md#framerate)          | String formatted as described in the XMP documentation |
+| [FrameCount](https://github.com/adobe/xmp-docs/blob/master/XMPNamespaces/XMPDataTypes/CoreProperties.md#framecount)        | String formatted as described in the XMP documentation |
+| [Part](https://github.com/adobe/xmp-docs/blob/master/XMPNamespaces/XMPDataTypes/CoreProperties.md#part)                    | String formatted as described in the XMP documentation |
+
+### Additional Properties
+
+| XMP Property Type | JSON Type |
+| ----------------- | --------- |
+| [ResourceEvent](https://github.com/adobe/xmp-docs/blob/master/XMPNamespaces/XMPDataTypes/ResourceEvent.md)| Object, we refer to the XMP documentation for a description of its properties |
+
+## Defining XMP Metadata
+
+An indirection level is introduced to avoid replicating the same XMP metadata in multiple glTF nodes.
+XMP metadatas are defined within a dictionary property in the glTF scene description file by adding an `extensions` property to the top-level glTF 2.0 object and defining a `KHR_xmp` object. The `KHR_xmp` object defines two properties:
+
+* `namespaces` : a dictionary mapping XMP [namespaces](https://github.com/adobe/xmp-docs/blob/master/Namespaces.md) to the URI where they are defined.
+* `packets`: an array of metadata packets referencing the namespaces defined in the `namespaces` dictionary.
+
+The following example defines a glTF scene with a sample XMP metadata.
+
+```json
+"extensions": {
+    "KHR_xmp" : {
+        "namespaces" : {
+            "dc" : "http://purl.org/dc/elements/1.1/"
+        },
+        "packets" : [
+            {
+                "dc:contributor" : [ "Creator1Name", "Creator2Email@email.com", "Creator3Name<Email@email.com>"],
+                "dc:coverage" : "Bay Area, California, United States",
+                "dc:creator" : [ "CreatorName", "CreatorEmail@email.com"],
+                "dc:date" : ["1997-07-16T19:20:30+01:00"],
+                "dc:description" : { "en-us": "text"},
+                "dc:format" : "model/gltf-binary",
+                "dc:identifier" : "urn:stock-id:292930",
+                "dc:language" : ["en"],
+                "dc:publisher" : ["Company"],
+                "dc:relation" : ["https://www.khronos.org/"],
+                "dc:rights" : {"en-us": "BSD"},
+                "dc:source" : "http://related_resource.org/derived_from_this.gltf",
+                "dc:subject" : ["architecture"],
+                "dc:title" : { "en-us" : "MyModel", "it-IT" : "Mio Modello"},
```

Comment:
Thanks for looking into this Don. Yes I think we should avoid cherry-picking.
If you think it would beneficial to the glTF community we could specify that a KHR_xmp processor has to enforce lower-case for language alternative keys. I don't see any drawback to it, what do you think?

Do you think the current wording is clear enough `A language alternative is a dictionary mapping a language (RFC 3066 language code) to text` or should we perhaps reference the xmp specification you posted ?

Thanks

Reply:
In the JSON-LD serialization of XMP (ISO 16684-3) which is what this work is based on, language codes are [BCP-47](https://tools.ietf.org/html/bcp47).  In section 2.1.1. of that standard, it is stated clearly:
>   At all times, language tags and their subtags, including private use
   and extensions, are to be treated as case insensitive: there exist
   conventions for the capitalization of some of the subtags, but these
   MUST NOT be taken to carry meaning.

Reply:
Because JSON objects are case-sensitive, this means it wouldn't be safe for an implementation to do lookups on the `{ "en-us" : ... } ` key. The object must be converted to an array, and iterated over doing case-insensitive checks on the key. Perhaps this should be an array, if we are not going to make the case predictable.

There is also language in the XML spec to suggest that `x-default` must always be the _first_ item in the alternative array. I'm not sure I understand the reasoning for that, or whether it applies to us with a JSON serialization.

Reply:
Good point @donmccurdy.  So make it a recommendation in your spec to make them lowercase.

It can't be an array since you need the object format to map the language to the value, in standard JSON.  In the full JSON-LD serialization, it uses the `@language` feature present there to handle this.

Reply:
`x-default` isn't present in the JSON-LD serialization.

Reply:
The [Core Properties](https://github.com/adobe/xmp-docs/blob/master/XMPNamespaces/XMPDataTypes/CoreProperties.md#language-alternative) language also says:

> A default value, if known, should be the first array item. The order of other array items is not specified by this document.

This requirement will be omitted from this extension, then,  and since this extension has no `@language` feature that information is lost. I don't have an opinion on whether that matters, but just want to confirm it.

Reply:
@donmccurdy I am confused about what information you think is lost.  The language is simply done via the key instead of `@language`